roles/maas_postgres_proxy/templates/haproxy.cfg.j2

```diff
@@ -0,0 +1,48 @@
+global
+    maxconn 100
+
+defaults
+    log global
+    mode tcp
+    retries 2
+    timeout client 30m
+    timeout connect 4s
+    timeout server 30m
+    timeout check 5s
+
+{% if maas_postgres_proxy_enabled %}
+    {% if (maas_proxy_postgres_bind_addr_v4 is defined) and (maas_proxy_postgres_bind_addr_v4) %}
+
+frontend maas-postgres-v4
+    mode tcp
+    bind {{ maas_proxy_postgres_bind_addr_v4 }}:{{ maas_proxy_postgres_port }}
+    use_backend maas-postgres
+
+    {% endif %}
+
+    {% if (maas_proxy_postgres_bind_addr_v6 is defined) and (maas_proxy_postgres_bind_addr_v6) %}
+
+frontend maas-postgres-v6
+    mode tcp
+    bind {{ maas_proxy_postgres_bind_addr_v6 }}:{{ maas_proxy_postgres_port }}
+    use_backend maas-postgres
+
+    {% endif %}
+
+backend maas-postgres
+    {% if maas_proxy_postgres_upstreams %}
+        {% for upstream in maas_proxy_postgres_upstreams %}
+    server {{ upstream['hostname'] }} {{ upstream['ip'] }}:{{ upstream['port'] }}
```

Comment:
this `backend` will require a set of `tcp-check`s for primary, Alternatively, we could create a postgres user specific to primaries and another specific to secondaries and use `psql-check <user>`.